syft/cataloger/rust/parse_cargo_lock_test.go

```diff
@@ -0,0 +1,99 @@
+package rust
+
+import (
+	"os"
+	"testing"
+
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/go-test/deep"
+)
+
+func TestParseCargoLock(t *testing.T) {
+	expected := []pkg.Package{
+		{
+			Name:     "ansi_term",
+			Version:  "0.12.1",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "matches",
+			Version:  "0.1.8",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "memchr",
+			Version:  "2.3.3",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "natord",
+			Version:  "1.0.9",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "nom",
+			Version:  "4.2.3",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "unicode-bidi",
+			Version:  "0.3.4",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "version_check",
+			Version:  "0.1.5",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
+		},
+		{
+			Name:     "winapi",
+			Version:  "0.3.9",
+			Language: pkg.Rust,
+			Type:     pkg.RustPkg,
+			Licenses: nil,
```

Comment:
is licenses not a thing with Rust packages? if it is, it would be useful to capture that in these tests

Reply:
No, there isn't any information about licenses available in the Cargo lock file

Reply:
This is an interesting question! I'm very new to Rust 😃 . From what I can tell, @westonsteimel is right about licenses not being stored in **cargo.lock** files. It looks like they are stored in **cargo.toml** files (akin to a `package.json` file), but that'd be a different parser than what's tackled here. We could handle that down the road.

[Cargo.toml vs Cargo.lock](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html)

[The Manifest Format (cargo.toml)](https://doc.rust-lang.org/cargo/reference/manifest.html)

Reply:
Yes, you can also use the [cargo metadata](https://doc.rust-lang.org/cargo/commands/cargo-metadata.html) command to output json with everything, but that would require executing a command